In [1]:
import numpy as np
from matplotlib import pyplot as plt

#configure plotting
%matplotlib inline
%config InlineBackend.figure_format = 'svg'
import matplotlib
matplotlib.rcParams['figure.figsize'] =  (8,5)
matplotlib.rcParams['text.usetex'] = True
matplotlib.rcParams['font.size'] = 16
matplotlib.rcParams['font.family'] = 'serif'

np.random.seed(0)


In [2]:
X = np.load('twoD_clustering_example.numpy')
print X.shape

(506, 2)


We have a two-dimensional dataset, with 506 data.

Building the model is as simple as importing the GPclust library and calling the class constructor with our data. The argument K specifies how many clusters to use.

In [4]:
import GPclust
m = GPclust.MOG(X, num_clusters=10)
print m.get_phi()

[[ 0.1899072   0.0053279   0.03517011 ...,  0.07245283  0.01314764
   0.40098088]
 [ 0.20134963  0.10387766  0.04287965 ...,  0.26966396  0.07398069
   0.05923953]
 [ 0.08490301  0.15517763  0.04202434 ...,  0.06451708  0.02897642
   0.28431159]
 ..., 
 [ 0.15220887  0.17539786  0.08623784 ...,  0.15292876  0.06494828
   0.03002559]
 [ 0.11378556  0.04689644  0.07583209 ...,  0.02729766  0.02676035
   0.01520369]
 [ 0.03033171  0.12708498  0.00995436 ...,  0.08080533  0.37354209
   0.11119468]]


In [5]:
from pprint import pprint

In [6]:
pprint (vars(m))

{'D': 2,
 'LOGPI': 1.1447298858494002,
 'S0': <tf.Tensor 'mul_1:0' shape=(2, 2) dtype=float64>,
 'S0_halflogdet': <tf.Tensor 'Sum_1:0' shape=() dtype=float64>,
 'X': <GPflow.param.DataHolder object at 0x11975d6d0>,
 'XXT': array([[[ 0.00083681, -0.00502729],
        [-0.00502729,  0.03020224]],

       [[ 0.00077919, -0.00489457],
        [-0.00489457,  0.03074588]],

       [[ 0.00050524, -0.00383388],
        [-0.00383388,  0.02909224]],

       ..., 
       [[ 0.0125053 , -0.0086217 ],
        [-0.0086217 ,  0.00594418]],

       [[ 0.01516977, -0.00987434],
        [-0.00987434,  0.00642743]],

       [[ 0.01399544, -0.01051532],
        [-0.01051532,  0.00790058]]]),
 '_free_vars': <tf.Tensor 'Placeholder_1:0' shape=<unknown> dtype=float64>,
 '_get_phi_AF_storage': {'free_vars': <tf.Tensor 'Placeholder_2:0' shape=(?,) dtype=float64>,
                         'session': <tensorflow.python.client.session.Session object at 0x11975d650>,
                         'tf_args': [],
       

In [7]:
import GPflow

In [8]:
# build a very simple data set:
N = 12
X = np.random.rand(N,1)
Y = np.sin(12*X) + 0.66*np.cos(25*X) + np.random.randn(N,1)*0.1 + 3

In [9]:
#build the GPR object
k = GPflow.kernels.Matern52(1, lengthscales=0.3)
meanf = GPflow.mean_functions.Linear(1,0)
m = GPflow.gpr.GPR(X, Y, k, meanf)
m.likelihood.variance = 0.01

In [10]:
pprint (vars(m))

{'X': <GPflow.param.DataHolder object at 0x1196f3fd0>,
 'Y': <GPflow.param.DataHolder object at 0x1196f3f90>,
 '_free_vars': <tf.Tensor 'Placeholder_3:0' shape=<unknown> dtype=float64>,
 '_name': 'model',
 '_needs_recompile': True,
 '_parent': None,
 '_session': <tensorflow.python.client.session.Session object at 0x119704790>,
 '_tf_mode': False,
 'kern': <GPflow.kernels.Matern52 object at 0x1196f3e10>,
 'likelihood': <GPflow.likelihoods.Gaussian object at 0x1196f3d50>,
 'mean_function': <GPflow.mean_functions.Linear object at 0x1196f3d90>,
 'num_latent': 1,
 'scoped_keys': ['build_likelihood', 'build_prior']}
